airflow/api_fastapi/core_api/datamodels/assets.py

```diff
@@ -99,3 +99,21 @@ class AssetEventCollectionResponse(BaseModel):
 
     asset_events: list[AssetEventResponse]
     total_entries: int
+
+
+class CreateAssetEventsBody(BaseModel):
+    """Create asset events request."""
+
+    uri: str
+    extra: dict = Field(default_factory=dict)
+
+    @field_validator("extra", mode="after")
+    def set_from_rest_api(cls, v):
+        if isinstance(v, dict):
+            v["from_rest_api"] = True
```

Comment:
Now we can add typing to the method because pydantic already validated that, so we are sure it's a dict at this point.

```suggestion
    def set_from_rest_api(cls, v: dict) -> dict:
        v["from_rest_api"] = True
```